### PR TITLE
feat: Batch Text Embeddings Sample

### PR DESCRIPTION
**Pull request by BigBlackWolf**
## Description

Sample for docs page https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/batch-prediction-genai-embeddings#generative-ai-batch-embedding-python_vertex_ai_sdk

As a reference was taken Python PR https://github.com/GoogleCloudPlatform/python-docs-samples/pull/12031

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: aiplatform/snippets/embedding_batch_predict.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	if err != nil {
+		return err
+	}
+
+	req := &aiplatformpb.CreateBatchPredictionJobRequest{
+		Parent: fmt.Sprintf("projects/%s/locations/%s", projectID, location),
+		BatchPredictionJob: &aiplatformpb.BatchPredictionJob{
+			DisplayName: name,
+			Model:       model,
+			InputConfig: &aiplatformpb.BatchPredictionJob_InputConfig{
```
Comment (telpirion) [inline aiplatform/snippets/embedding_batch_predict.go:50]: nit: consider initializing the InputConfig and OutputConfig outside of the request.

**Review by telpirion — APPROVED** [review]

File: aiplatform/snippets/embedding_batch_predict.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	// outputURI: existing template path. Following formats are allowed:
+	// 	- gs://BUCKET_NAME/DIRECTORY/
+	// 	- bq://project_name.llm_dataset
+
+	ctx := context.Background()
+	apiEndpoint := fmt.Sprintf("%s-aiplatform.googleapis.com:443", location)
+	// Pretrained model
+	model := "publishers/google/models/textembedding-gecko"
+
+	client, err := aiplatform.NewJobClient(ctx, option.WithEndpoint(apiEndpoint))
```
Comment (telpirion) [inline aiplatform/snippets/embedding_batch_predict.go:40]: Does this client need to be closed?

**Review by telpirion — DISMISSED** [review]

File: aiplatform/snippets/embedding_batch_predict.go (hunk 1 above)
Comment (BigBlackWolf) [reply aiplatform/snippets/embedding_batch_predict.go]: Updated

File: aiplatform/snippets/embedding_batch_predict.go (hunk 2 above)
Comment (BigBlackWolf) [reply aiplatform/snippets/embedding_batch_predict.go:59]: Good catch, I missed it. Updated with closing

**Review by telpirion — APPROVED** [review]

**Pull request merged**